pennylane/pauli/pauli_arithmetic.py

```diff
@@ -414,10 +440,61 @@ def _pw_wires(w: Iterable) -> wires.Wires:
 
         return math.expand_matrix(reduced_mat, result_wire_order, wire_order=wire_order)
 
+    def _to_sparse_mat(self, wire_order):
+        """Compute the sparse matrix of the Pauli sentence by efficiently adding the Pauli words
+        that conform it. See pauli_sparse_matrices.md for the technical details."""
+        pauli_words = list(self)  # Ensure consistent ordering
+        n_wires = len(wire_order)
+        matrix_size = 2**n_wires
+        matrix = sparse.csr_matrix((matrix_size, matrix_size), dtype="complex128")
+        op_sparse_idx = _ps_to_sparse_index(pauli_words, wire_order)
+        _, unique_inds, unique_invs = np.unique(
+            op_sparse_idx, axis=0, return_index=True, return_inverse=True
+        )
+        pw_sparse_structures = unique_inds[unique_invs]
+
+        buffer_size = 2 ** max(0, 31 - n_wires - 5)  # Buffer of 1GB memory
+        mat_data = np.empty((matrix_size, buffer_size), dtype=np.complex128)
+        mat_indices = np.empty((matrix_size, buffer_size), dtype=np.int64)
+        i = 0
+        for sparse_structure in unique_inds:
+            idx = np.where(pw_sparse_structures == sparse_structure)[0]
```

Comment:
Is `idx` a single index or an iterable of indices?  Why do we need to select out `[0]`?

Reply:
I have changed this for `np.nonzero`, which is actually what's being called inside `np.where`. This returns a tuple with the nonzero arrays for each dimension, hence the indexing. However, I have changed now renaming the variable and untupling explicitly. I think it's much clearer now 😄